### PR TITLE
kGH-21: Add BinderHeaders.PARTITION_OVERRIDE

### PR DESCRIPTION
**Pull request by garyrussell**
See spring-cloud/spring-cloud-stream-binder-kafka#109
See #851

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderHeaders.java [hunk 1]
```diff
@@ -40,6 +40,12 @@
 	public static final String PARTITION_HEADER = "partition";
 
 	/**
+	 * Indicates the target partition of an outbound message. Overrides
+	 * any partition selected by the binder.
+	 */
+	public static final String PARTITION_OVERRIDE = "partitionOverride";
```
Comment (artembilan) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderHeaders.java]: Not related to the subject, but why there is no `prefix`?
Imagine downstream application which has some logic around headers.
So, how to distinguish binder headers from all others?
Or just how to understand that the header in target message is related to binder not some other "hell"?

Thanks

Comment (garyrussell) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderHeaders.java]: Yes; I thought about that too but I fear it's too late now 😦

Comment (artembilan) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderHeaders.java]: Well, we always can do that and support both headers sets some time.
Like that was once in Spring Cloud Sleuth.

**Comment by garyrussell** [comment]
Talked to Marius; we decided it's ok to prefix the partition headers but not the oCT - testing now...

**Comment by mbogoevici** [comment]
@garyrussell Small updates to Javadoc - let me know if you agree. https://github.com/mbogoevici/spring-cloud-stream/commit/458ed08688a483d798f25bb3698e3bfeff73f42c

**Comment by garyrussell** [comment]
LGTM; thanks.

**Comment by mbogoevici** [comment]
Squashed and merged w/updates via https://github.com/spring-cloud/spring-cloud-stream/commit/bd1454bffff52d29f113f040076eb8fc656122cf

**Pull request closed without merging**